src/vs/workbench/services/lifecycle/electron-sandbox/lifecycleService.ts

```diff
@@ -87,7 +88,11 @@ export class NativeLifecycleService extends AbstractLifecycleService {
 		}, NativeLifecycleService.BEFORE_SHUTDOWN_WARNING_DELAY);
 
 		await Promise.race([
-			this.remoteAgentService.endConnection(),
+			this.remoteAgentService.endConnection().catch(e => {
```

Comment:
It seems we got back a dependency of lifecycle to remote, which I find the wrong direction. I recall we added an order property to the shutdown event:

https://github.com/microsoft/vscode/blob/ed3e34790983712fa73f93ddc753712530c73ad5/src/vs/workbench/services/extensions/common/abstractExtensionService.ts#L200-L204

If we cannot use it here, maybe we can find something that catches all cases and allows us to break this odd dependency?

Reply:
Ugh, that seems to have come in with https://github.com/microsoft/vscode/pull/219041. Somehow that branch included an old revision of the disconnect logic instead of its intended commit. I'll pull that out in this PR 😬

Reply:
Fortunately because the real PR was merged first that ended up being a pretty minimal commit that just effected the changed we backed out from the lifecycle service